include/odp/api/spec/packet.h

```diff
@@ -1979,6 +1979,23 @@ int8_t odp_packet_shaper_len_adjust(odp_packet_t pkt);
  */
 void odp_packet_shaper_len_adjust_set(odp_packet_t pkt, int8_t adj);
 
+/**
+ * Get mark id value from the odp_packet_t
+ *
+ * When a flow configured with mark CoS, Upon hitting the mark CoS,
```

Comment:
Maybe we should not discuss the classifier details too much in here as from packet point of view this just somewhat arbitrary metadata that can perhaps be used in more ways later (like allowing the application set the value directly).

I would reduce the explanation to something shorter, like this, but add a comment about the initial value in case CoS is not even involved (e.g. someone just allocs a packet and later reads the CoS mark):

"CoS class can be set by the classifier to an application defined value. The default value is ODP_COS_MARK_INVALID."

Reply:
I will wait for your comments on new API proposal as odp_cls_pmr_create_with_mark() to reword the same.